File: beecrowd/python/estruturas-e-bibliotecas/ex1110.py
def realocar(cartas):
    last = [cartas.pop()]
    last.extend(cartas)
    return last

while True:
    n = int(input())
    if n == 0: break
    
    cartas = [i for i in range(1, n+1)]
    cartas.reverse()
    
    saida = 'Discarded cards: '
    while len(cartas) > 1:
        if len(cartas) > 2: saida += f'{cartas.pop()}, ' 
        else: saida += str(cartas.pop())
        cartas = realocar(cartas)
    saida += f'\nRemaining card: {cartas.pop()}'
    print(saida)

File: beecrowd/c++/estruturas-e-bibliotecas/ex1110.cpp
#include <iostream>
#include <queue>
#include <vector>
using namespace std;

int main(){
    int n;
    scanf("%d", &n);
    while (n != 0){
        queue<int> fila;
        vector<int> discard;

        for(int i=1; i<=n; i++){
            fila.push(i);
        }

        while (!fila.empty()){
            discard.push_back(fila.front());
            fila.pop();
            fila.push(fila.front());
            fila.pop();
        }

        cout<<"Discarded cards: ";
        for(int i=0; i<n-1; i++){
            cout<<discard[i];
            if (i!=n-2){
                cout<<", ";
            }
        }

        cout<<"\nRemaining card: "<<discard[n-1]<<endl;

        scanf("%d", &n);
    }
}